casper/src/main/scala/coop/rchain/casper/genesis/contracts/Faucet.scala

```diff
@@ -0,0 +1,24 @@
+package coop.rchain.casper.genesis.contracts
+
+import coop.rchain.rholang.mint.BasicWalletFaucet
+
+/**
+  * The purpose of a "Faucet" is to give a place where users
+  * can obtain REV for testing their contracts for free on the
+  * testnet. Since this is only applicable to testnet, the
+  * `noopFaucet` will be used in main net, where there is no
+  * way to obtain REV for free.
+  */
```

Comment:
Will the faucet disappear once we start charging from the real wallets?

Reply:
If you look above there is a boolean flag that is switched on for testnet/testing and off for when it is mainnet.

Reply:
@goral09 Part of the reason for having the faucet is because we will be charging from wallets soon. We still want people to be able to try out the system even once we start charging, so they will be able to by getting a wallet from the faucet. But as @KentShikama says, of course we won't have this in main net where the REV actually represents something of value.

Reply:
What I meant by asking this question was if on the main net we will replace faucets with the real wallets.

Reply:
Ah, yes, absolutely. And we even already have the infrastructure for that; it's the `walletsFile` that defines all the REV which will be available in main net.